core/modules/rproc/InfileMerger.cc

```diff
@@ -298,16 +311,32 @@ bool InfileMerger::isFinished() const {
 }
 
 
-bool InfileMerger::_deleteInvalidRows(int jobIdAttempt) {
-    std::string sqlDelRows = std::string("DELETE FROM ") + _mergeTable
-                             + " WHERE " +  _jobIdColName + "=" + std::to_string(jobIdAttempt);
-    LOGS(_log, LOG_LVL_DEBUG, "Deleting invalid rows w/" << sqlDelRows);
-    bool ok = _applySqlLocal(sqlDelRows, "deleteInvalidRows");
-    if (!ok) {
-        LOGS(_log, LOG_LVL_ERROR, "Failed to drop columns w/" << sqlDelRows);
-        return false;
+bool InfileMerger::_deleteInvalidRows(std::set<int> const& jobIdAttempts) {
+    // delete several rows at a time
+    unsigned int maxSize = 950000; /// default 1mb limit
```

Comment:
Why is this magic limit (effectively for a total number of invalid jobs to be deleted at once) here? This is perhaps an attempt to deal with some "future" problem which will never occur. From my point of view it just reduces the overall clarity of the code. Would it be possible to refactor the code to set a limit on a number of jobs to be removed with a single DELETE statement and then add another loop to iterate over all jobs while it's all done.

And if there is a concern to run into the "run-away" scenario then reporting each iteration into the log stream may help to detect & debug the problem.